src/kex.c

```diff
@@ -98,6 +98,44 @@
             }                                                              \
     }
 
+/*
+ * Generic Diffie-Hellman computation support.
+ *
+ * DH context should be a _libssh2_bn *.
+ */
+
+#ifndef libssh2_dh_key_pair
```

Comment:
I would rather merge your current tested code, even if it's not perfect.  We can always clean it up in the future should you regain access to OS400.

Could you fix the typos and then i'll merge it as it is.